fabtests/unit/av_test.c

```diff
@@ -825,12 +825,206 @@ av_lookup_bad(void)
 	return TEST_RET_VAL(ret, testret);
 }
 
+static int
+av_remove_good(void)
+{
+	int testret, ret, i;
+	struct fid_av *av = NULL;
+	struct fi_av_attr attr;
+	uint8_t addrbuf[4096];
+	size_t buflen;
+	fi_addr_t *fi_addr;
+
+	testret = FAIL;
+
+	fi_addr = calloc(num_good_addr > 0 ? num_good_addr : 1,
+			 sizeof(fi_addr));
+	if (!fi_addr) {
+		sprintf(err_buf, "malloc fi_addr failed");
+		ret = -FI_ENOMEM;
+		goto fail;
+	}
+
+	memset(&attr, 0, sizeof(attr));
+	attr.type = av_type;
+	attr.count = num_good_addr;
+
+	ret = fi_av_open(domain, &attr, &av, NULL);
+	if (ret != 0) {
+		sprintf(err_buf, "fi_av_open(%s) = %d, %s",
+			fi_tostr(&av_type, FI_TYPE_AV_TYPE), ret,
+			fi_strerror(-ret));
+		goto fail;
+	}
+
+	for (i = 0; i < num_good_addr; i++)
+		fi_addr[i] = FI_ADDR_NOTAVAIL;
+
+	buflen = sizeof(addrbuf);
+	ret = av_create_address_list(good_address, 0, num_good_addr, addrbuf, 0,
+				     buflen);
+	if (ret < 0)
+		goto fail;
+
+	ret = fi_av_insert(av, addrbuf, num_good_addr, fi_addr, 0, NULL);
+	if (ret != num_good_addr) {
+		sprintf(err_buf, "fi_av_insert ret=%d, expected %d", ret,
+			num_good_addr);
+		goto fail;
+	}
+
+	for (i = 0; i < num_good_addr; i++) {
+		if (fi_addr[i] == FI_ADDR_NOTAVAIL) {
+			sprintf(err_buf, "fi_addr[%d] expected %ld, actual %ld",
+				i, FI_ADDR_NOTAVAIL, fi_addr[i]);
+			goto fail;
+		}
+	}
+
+	for (i = 0; i < num_good_addr; i++) {
+		ret = fi_av_remove(av, &fi_addr[i], 1, 0);
+		if (ret) {
+			sprintf(err_buf, "fi_av_remove ret=%d, %s", ret,
+				fi_strerror(-ret));
+			goto fail;
+		}
+	}
+
+	for (i = 0; i < num_good_addr; i++) {
+		if (fi_addr[i] != FI_ADDR_NOTAVAIL) {
+			sprintf(err_buf, "fi_av_remove failed fi_addr[%d] "
+				"expected %ld, actual %ld", i, FI_ADDR_NOTAVAIL,
+				fi_addr[i]);
+			goto fail;
+		}
+	}
```

Comment:
What is this checking? Does this pass any provider? fi_av_remove doesn't set the inputted fiaddr to FI_ADDR_NOTAVAIL. They will be the same value, just not valid anymore